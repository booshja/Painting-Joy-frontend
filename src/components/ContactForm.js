// dependencies
import React from "react";
import { useHistory } from "react-router";
import { useForm } from "react-hook-form";
import styled from "styled-components";
import axios from "axios";

const StyledForm = styled.form`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
`;

const StyledLabel = styled.label`
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    letter-spacing: 1px;
    margin-bottom: 0.25rem;
`;

const StyledInput = styled.input`
    border: 2px solid #207070;
    border-radius: 4px;
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    letter-spacing: 1px;
    margin-bottom: 0.5rem;
`;

const StyledField = styled.textarea`
    border: 2px solid #207070;
    border-radius: 4px;
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    letter-spacing: 1px;
    margin-bottom: 1rem;
`;

const StyledButton = styled.button`
    background-color: #207070;
    border: none;
    border-radius: 4px;
    padding: 2.5px 10px;
    color: #ffffff;
    font-family: "News Cycle", sans-serif;
    font-weight: 700;
    font-size: 1.4rem;
    letter-spacing: 1px;
    align-self: flex-end;

    &::hover {
        background-color: #ffffff;
        color: #207070;
    }
`;

const StyledError = styled.p`
    color: #990000;
    font-family: "News Cycle", sans-serif;
    font-weight: 700;
    font-size: 1.2rem;
    letter-spacing: 1px;
    padding: 0.5rem;
`;

const ContactForm = () => {
    // TODO: HANDLE POST REQUEST
    const {
        register,
        handleSubmit,
        formState: { errors },
    } = useForm();
    const history = useHistory();

    const handleDataSubmit = async (data) => {
        try {
            const res = await axios.post(
                process.env.REACT_APP_BACKEND_URL + "/messages",
                data
            );
            console.log("RESSSS", res);
        } catch (err) {
            console.log("ERRORRRRR", err);
        }
    };
    return (
        <StyledForm
            onSubmit={handleSubmit((data) => {
                handleDataSubmit(data);
            })}
        >
            <StyledLabel htmlFor="name">Name:</StyledLabel>
            <StyledInput
                type="text"
                placeholder="Name"
                id="name"
                {...register("name", { required: "Name is required" })}
            />
            {errors.name && <StyledError>{errors.name.message}</StyledError>}
            <StyledLabel htmlFor="email">Email:</StyledLabel>
            <StyledInput
                type="text"
                placeholder="Email:"
                id="email"
                {...register("email", { required: "Email is required" })}
            />
            {errors.email && <StyledError>{errors.email.message}</StyledError>}
            <StyledLabel htmlFor="message">Message:</StyledLabel>
            <StyledField
                placeholder="Message"
                id="message"
                {...register("message", {
                    required: "Message is required",
                    maxLength: {
                        value: 200,
                        message: "Maximum message length 200 characters",
                    },
                })}
            />
            {errors.message && (
                <StyledError>{errors.message.message}</StyledError>
            )}
            <StyledButton type="submit">Submit</StyledButton>
        </StyledForm>
    );
};

export default ContactForm;
