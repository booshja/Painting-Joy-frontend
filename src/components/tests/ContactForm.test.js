import { render } from "@testing-library/react";
import ContactForm from "../ContactForm";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing", () => {
    render(<ContactForm />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByLabelText, getByRole } = render(<ContactForm />);

    expect(getByLabelText("Name:")).toBeInTheDocument();
    expect(getByLabelText("Email:")).toBeInTheDocument();
    expect(getByLabelText("Message:")).toBeInTheDocument();
    expect(getByRole("button", { name: "Submit" })).toBeInTheDocument();
});
