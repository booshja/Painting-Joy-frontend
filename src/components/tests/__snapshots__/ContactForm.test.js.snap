// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-bottom: 0.25rem;
}

.c2 {
  border: 2px solid #207070;
  border-radius: 4px;
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-bottom: 0.5rem;
}

.c3 {
  border: 2px solid #207070;
  border-radius: 4px;
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-bottom: 1rem;
  resize: none;
}

.c4 {
  background-color: #207070;
  border: none;
  border-radius: 4px;
  padding: 0.5px 8px;
  color: #ffffff;
  font-family: "News Cycle",sans-serif;
  font-weight: 700;
  font-size: 1.4rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  cursor: pointer;
  border: 2px solid #207070;
}

.c4:hover {
  background-color: #ffffff;
  color: #207070;
}

@media screen and (min-width:414px) {
  .c0 {
    width: 90%;
  }
}

@media screen and (min-width:414px) {
  .c2 {
    width: 95%;
  }
}

@media screen and (min-width:414px) {
  .c3 {
    width: 95%;
  }
}

@media screen and (min-width:414px) {
  .c4 {
    margin-right: 0.7rem;
  }
}

<form
    class="c0"
  >
    <div />
    <label
      class="c1"
      for="name"
    >
      Name:
    </label>
    <input
      class="c2"
      id="name"
      name="name"
      placeholder="Name"
      type="text"
    />
    <label
      class="c1"
      for="email"
    >
      Email:
    </label>
    <input
      class="c2"
      id="email"
      name="email"
      placeholder="Email:"
      type="text"
    />
    <label
      class="c1"
      for="message"
    >
      Message:
    </label>
    <textarea
      class="c3"
      id="message"
      maxlength="200"
      name="message"
      placeholder="Message"
      rows="4"
    />
    <button
      class="c4"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
